test/v4/hooks/dca/DCAHook_transferInputTokens.t.sol

```diff
@@ -325,6 +325,47 @@ contract DCAHook_transferInputTokensTest is Test, DeployPermit2 {
         hook.transferInputTokens(order, FILLER, permitData);
     }
 
+    function test_transferInputTokens_withPermit_frontRunProtection() public {
+        // Get the current nonce for the swapper
+        (,, uint48 currentNonce) = permit2.allowance(SWAPPER, address(inputToken), address(hook));
+
+        // Create permit with proper signature
+        IAllowanceTransfer.PermitSingle memory permitSingle = IAllowanceTransfer.PermitSingle({
+            details: IAllowanceTransfer.PermitDetails({
+                token: address(inputToken),
+                amount: uint160(AMOUNT),
+                expiration: uint48(block.timestamp + 1000),
+                nonce: currentNonce
+            }),
+            spender: address(hook),
+            sigDeadline: block.timestamp + 1000
+        });
+
+        bytes memory signature = _generatePermitSignature(permitSingle);
+        PermitData memory permitData = PermitData({hasPermit: true, permitSingle: permitSingle, signature: signature});
+        ResolvedOrder memory order = _createResolvedOrder(SWAPPER, address(inputToken), AMOUNT);
+
+        // ATTACK: Front-runner extracts the permit from mempool and calls permit2.permit() directly
+        // This consumes the nonce but sets the allowance for the hook (as specified in the permit)
+        permit2.permit(SWAPPER, permitSingle, signature);
+
+        // Verify the nonce was consumed by the front-runner
+        (,, uint48 nonceAfterFrontRun) = permit2.allowance(SWAPPER, address(inputToken), address(hook));
+        assertEq(nonceAfterFrontRun, currentNonce + 1);
+
+        // PROTECTED: User's transaction should still succeed despite front-run
+        // Before fix: would revert with "invalid nonce"
```

Comment:
Nit: I wouldn't mention a "fix" in the comments

Reply:
`// Because try-catch catches the permit failure, transfer succeeds with existing allowance`